nodejs/eks/examples/storage-classes/utils.ts

```diff
@@ -0,0 +1,22 @@
+import * as eks from "@pulumi/eks";
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+
+export function checkStorageClass(storageClasses: eks.UserStorageClasses) {
+    if (storageClasses) {
+        pulumi.output(storageClasses).apply(scs => {
+            for (const key of Object.keys(scs)) {
+                const sc = scs[key];
+                sc.metadata.apply(m => {
+                    // 'defaultName' used per suggestion in https://git.io/fjVYi due to
+                    // https://github.com/pulumi/pulumi/issues/2433
+                    const name = m.name || "defaultName";
+                    if (!name) {
```

Comment:
Wouldn't this always evaluate to `true`? If `m.name` is falsy, I think `defaultName` would be assigned.

Reply:
This was done per the suggestion in #2 of https://git.io/fjVYi due to the [issue](https://github.com/pulumi/pulumi/issues/2433) linked in the comment above it that returns an `undefined` on the first `pulumi preview`. During a preview this will show up as `defaultName`, but after `pulumi update --skip-preview` we get the right storageClass names:

ex. on `preview`:
![image](https://user-images.githubusercontent.com/1112768/59812997-c94ca700-92c4-11e9-92de-540fdda8cd6d.png)

but then on the `update`:
![image](https://user-images.githubusercontent.com/1112768/59812981-bc2fb800-92c4-11e9-85cf-d33a184e6ef5.png)

Reply:
I realized I had that backwards. I don't think the conditional block will ever execute since the `||` guarantees that a truthy value will be set.

Reply:
Yes - @lblackstone is right - this conditional looks like it can never be true and this test will never fail.